lib/_stream_readable.js

```diff
@@ -8,27 +8,30 @@ const Stream = require('stream');
 const Buffer = require('buffer').Buffer;
 const util = require('util');
 const debug = util.debuglog('stream');
+const BufferList = require('internal/streams/BufferList');
 var StringDecoder;
 
 util.inherits(Readable, Stream);
 
-const hasPrependListener = typeof EE.prototype.prependListener === 'function';
-
-function prependListener(emitter, event, fn) {
-  if (hasPrependListener)
+var prependListener;
+if (typeof EE.prototype.prependListener === 'function') {
+  prependListener = function prependListener(emitter, event, fn) {
     return emitter.prependListener(event, fn);
-
-  // This is a brutally ugly hack to make sure that our error handler
-  // is attached before any userland ones.  NEVER DO THIS. This is here
-  // only because this code needs to continue to work with older versions
-  // of Node.js that do not include the prependListener() method. The goal
-  // is to eventually remove this hack.
-  if (!emitter._events || !emitter._events[event])
-    emitter.on(event, fn);
-  else if (Array.isArray(emitter._events[event]))
-    emitter._events[event].unshift(fn);
-  else
-    emitter._events[event] = [fn, emitter._events[event]];
+  };
+} else {
+  prependListener = function prependListener(emitter, event, fn) {
+    // This is a brutally ugly hack to make sure that our error handler
```

Comment:
If the comment here is being touched anyway, I wouldn’t mind removing `brutally ugly` from it.